Src/ST-LIB_LOW/Sensors/EncoderSensor/EncoderSensor.cpp

```diff
@@ -0,0 +1,43 @@
+#include "ST-LIB_LOW/Sensors/EncoderSensor/EncoderSensor.hpp"
+#include "Encoder/Encoder.hpp"
+#include "Time/Time.hpp"
+
+EncoderSensor::EncoderSensor(Pin pin1, Pin pin2, double counter_distance, double *position, double *speed, double *acceleration)
+: counter_distance(counter_distance), position(position), speed(speed), acceleration(acceleration){
+	optional<uint8_t> identification = Encoder::register_encoder(pin1,pin2);
+	if(identification){
+		id = identification.value();
+		Encoder::turn_on_encoder(id);
+		last_check = Time::get_global_tick();
+		last_position = *position;
+		last_speed = *speed;
+	}else{
+		//errores de optional vacio aqui (por hacer)
```

Comment:
Cuanndo hagas estos comentarios incidcando que queda algo por hacer añade un TODO: delante. Así despues el Cube te indica todos los TODO que te faltan y no te dejas ninguno.
```suggestion
		//TODO: Error handler, Error during register of encoder x
```